src/detect.rs

```diff
@@ -147,7 +147,7 @@ struct TrapFrame {
 // handle exceptions only rather than interrupts.
 #[naked]
 unsafe extern "C" fn on_detect_trap() -> ! {
-    asm!(
+    core::arch::naked_asm!(
```

Comment:
From nightly 2024-10-07 we should use `naked_asm!` instead of `asm!` in `#[naked]` function context. Ref: https://github.com/rust-lang/rust/pull/128651

Reply:
This is pretty awkward situation, we'd apprepriate it if you could offer some advice.